linker-private-library/src/main/scala/org/scalajs/linker/runtime/RuntimeLong.scala

```diff
@@ -1087,77 +1087,358 @@ object RuntimeLong {
+   *   it holds that |⌊◦(x)⌋ - ⌊y⌋| <= 1.
+   *
+   *   Proof. Observe that, in the given range, all multiples of 1/2 are
+   *   representable as doubles. Rewrite x = n + f with n an integer and
+   *   0 <= f < 1. Then x - 1/2 = n + f - 1/2 <= y <= n + f + 1/2 = x + 1/2.
+   *
+   *   If 0 <= f < 1/2, then n <= ◦(x) <= n + 1/2 and ⌊◦(x)⌋ = n.
+   *   n - 1/2 <= y < n + 1, so n - 1 <= ⌊y⌋ <= n.
+   *   Therefore |⌊◦(x)⌋ - ⌊y⌋| <= 1, as desired.
+   *
+   *   Otherwise, 1/2 <= f < 1, then n + 1/2 <= ◦(x) <= n + 1 and
+   *   n <= ⌊◦(x)⌋ <= n + 1. n <= y < n + 3/2, so n <= ⌊y⌋ <= n + 1.
+   *   Therefore |⌊◦(x)⌋ - ⌊y⌋| <= 1 as well, as desired.
+   *
+   * Now, we will show that
+   *
+   *   |(â / b̂) - (a / b)| <= 1/2      (1)
+   *
+   * To prove (1), we split into two subcases:
+   * either 2²¹ <= b < 2⁵³, or 2⁵³ <= b < 2⁶².
+   *
+   * If 2²¹ <= b < 2⁵³, we know that b̂ = b. Therefore,
+   *
+   *   |(â / b̂) - (a / b)|
+   *     = |((a + δa) / b) - (a / b)|
+   *     = |δa / b|
+   *     <= 2¹⁰/2²¹
+   *     < 1/2
+   *
+   * The subcase 2⁵³ <= b < 2⁶² is harder, as both â and b̂ can be inexact.
+   * We decompose
+   *
+   *   |(â / b̂) - (a / b)|
+   *     = |((a + δa) / (b + δb)) - (a / b)|
+   *     = |(b∙(a + δa) - (b + δb)∙a) / (b∙(b + δb))|
+   *     <= |(b∙(a + 2¹⁰) - (b - 2¹⁰)∙a) / (b∙(b - 2¹⁰))|
+   *       (taking extreme values of δa and δb to maximize the result)
+   *     = |(a∙b + b∙2¹⁰ - a∙b + a∙2¹⁰) / (b² - b∙2¹⁰)|
+   *     = |((a + b)∙2¹⁰) / (b² - b∙2¹⁰)|
+   *     <= |((2⁶⁴ + 2⁶²)∙2¹⁰) / ((2⁵³)² - 2⁵³∙2¹⁰)|
+   *       (taking extreme values of a and b to maximize the result)
+   *     ~= 2.91 * 10⁻¹⁰
+   *     < 1/2
+   *
+   * From (1), Lemma 1, and the fact that both (â / b̂) and (a / b)
+   * are <= 2⁴³ < 2⁵², we conclude that
+   *
+   *   |q̂ - q| = |⌊◦(â / b̂)⌋ - ⌊a/b⌋| <= 1
+   *
+   * as desired.
+   *
+   * ---
+   *
+   * Now that we have computed q̂ and established that |q̂ - q| <= 1, we can use
+   * the estimated remainder to fix the result.
+   *
+   * We compute
+   *
+   *   r̂ = a - b∙q̂
+   *
+   * and compare it to the exact remainder r = rem(a, b):
+   *
+   *   |r̂ - r|
+   *     = |(a - b∙q̂) - (a - b∙q)|
+   *     = |-b∙q̂ + b∙q|
+   *     = |b∙q̂ - b∙q|
+   *     = |b∙(q̂ - q)|
+   *     = b∙|q̂ - q|
+   *     <= b
+   *
+   * Combining with 0 <= r < b, we get -b <= r̂ < 2∙b. Since, b < 2⁶², we have
+   * -2⁶² < r̂ < 2⁶³, and therefore the computation of r̂ does not overflow when
+   * done with `long` operations, interpreting r̂ as a signed quantity.
+   *
+   * Finally, we can compare r̂ against 0 and against b to decide whether
+   * q̂ - q was -1, 0, or 1, and correct accordingly.
+   */
+
   /** Helper for `unsigned_/` and `unsigned_%`.
    *
    *  If `askQuotient` is true, computes the quotient, otherwise computes the
    *  remainder. Stores the hi word of the result in `hiReturn`, and returns
    *  the lo word.
    */
+  @inline // inlined twice; specializes for askQuotient
```

Comment:
Are you sure the increase code size is worth it for the speed? (IIUC this is why we'd inline here).

Or do we actually get smaller code with inlining?